Satisfy Charmhub requirements

Gleland opened the pull request:
add charmcraft and metadata changes to build with charmcraft pack successfully

Gleland committed:
Update metadata.yaml to support focal

Gleland committed:
Add charmcraft.yaml

satisfy requirements to build with charmcraft on launchpad.

afreiberger commented on charmcraft.yaml:
```diff
@@ -0,0 +1,18 @@
+              - amd64
+parts:
+  charm:
+    charm-python-packages: [setuptools < 58]
```
why is this necessary?  I don't see jinja2 or charmhelpers in requirements.txt

Gleland replied:
thanks! i've updated this, and the list/indentation usage to be consistent as well.

Gleland committed:
remove setuptools pinning

add indentation and explicit list usage to be consistent.
remove setuptools version pinning that isn't required
